go-controller/pkg/util/util.go

```diff
@@ -390,3 +390,19 @@ func NoHostSubnet(node *v1.Node) bool {
 	nodeSelector, _ := metav1.LabelSelectorAsSelector(config.Kubernetes.NoHostSubnetNodes)
 	return nodeSelector.Matches(labels.Set(node.Labels))
 }
+
+// GetNBZone returns the zone name configured in the OVN Northbound database.
+// If the zone name is not configured, it returns the default zone name - "global"
+func GetNBZone(nbClient libovsdbclient.Client) string {
+	nbGlobal := &nbdb.NBGlobal{}
+	nbGlobal, err := libovsdbops.GetNBGlobal(nbClient, nbGlobal)
+	if err != nil {
```

Comment:
Let's make this logic smarter, if possible. It should detect cases when there is a DB failure vs cases where the global is not present. I would go as far as making GetNBZone return (string, err) and make caller handle failures... or ignore them only when it is safe to do so.

Reply:
Ack